SamcoEnhanced/SamcoEnhanced.ino

```diff
@@ -1955,60 +2000,43 @@ void ExecCalMode()
     Serial.printf("CalStage: %d\r\n", Cali_Verify+1);
 }
 
+
 // Locking function moving from cali point to point
 void CaliMousePosMove(uint8_t caseNumber)
 {
-    int32_t xPos;
-    int32_t yPos;
-
-    // Note: All TinyUSB device updates are blocking when a previous signal hasn't finished transmitting,
-    // so we skip chunks to speed things the hell up.
-    // These originally were inc/decrementing by 1 with a 20us delay between.
-
+    // Set the target positions based on the calibration stage
     switch(caseNumber) {
         case Cali_Top:
-          for(yPos = 32768/2; yPos > 0; yPos = yPos - 30) {
-              if(yPos < 31) { yPos = 0; }
-              AbsMouse5.move(32768/2, yPos);
-          }
-          delay(5);
-          break;
+            mouseTargetX = 32768 / 2;
+            mouseTargetY = 0;
+            mouseMoving = true;
+            break;
         case Cali_Bottom:
-          for(yPos = 0; yPos < 32767; yPos = yPos + 30) {
-              if(yPos > 32767-31) { yPos = 32766; }
-              AbsMouse5.move(32768/2, yPos);
-          }
-          delay(5);
-          break;
+            mouseTargetX = 32768 / 2;
+            mouseTargetY = 32767;
+            mouseMoving = true;
+            break;
         case Cali_Left:
-          yPos = 32767;
-          for(xPos = 32768/2; xPos > 0; xPos = xPos - 30) {
-              if(xPos < 31) { xPos = 0; }
-              if(yPos > 32768/2) { yPos = yPos - 30; }
-              else { yPos = 32768/2; }
-              AbsMouse5.move(xPos, yPos);
-          }
-          delay(5);
-          break;
+            mouseTargetX = 0;
+            mouseTargetY = 32768 / 2;
+            mouseMoving = true;
+            break;
         case Cali_Right:
-          for(xPos = 0; xPos < 32767; xPos = xPos + 30) {
-              if(xPos > 32767-31) { xPos = 32766; }
-              AbsMouse5.move(xPos, 32768/2);
-          }
-          delay(5);
-          break;
+            mouseTargetX = 32767;
+            mouseTargetY = 32768 / 2;
+            mouseMoving = true;
+            break;
         case Cali_Center:
-          for(xPos = 32767; xPos > 32768/2; xPos = xPos - 30) {
-              if(xPos < 32768/2) { xPos = 32768/2; }
-              AbsMouse5.move(xPos, 32768/2);
-          }
-          delay(5);
-          break;
+            mouseTargetX = 32768 / 2;
+            mouseTargetY = 32768 / 2;
+            mouseMoving = true;
+            break;
         default:
-          break;
+            break;
     }
 }
 
```

Comment:
I know a lot of this is code I originally made, but if the only thing this function does is set init values, should we just put all these up in the ExecCalMode switch cases (after the cali values are set)? In retrospect, having this function where it is in the Calibration loop keeps tripping me up. :sweat_smile:

Reply:
Yeah, good point. The updated request has merged these switch cases into ExecCalMode.